<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Frame</class>
 <widget class="QFrame" name="Frame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>820</width>
    <height>372</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::StyledPanel</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Raised</enum>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
   <property name="title">
    <string>FiniteTrackCartPole</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>20</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
   <property name="title">
    <string>算法</string>
   </property>
   <widget class="QListWidget" name="lstAgent">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>151</width>
      <height>131</height>
     </rect>
    </property>
    <property name="currentRow">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>QLearning</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Sarsa</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="btnConfigAgent">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>参数设置</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>20</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
   <property name="title">
    <string>表示</string>
   </property>
   <widget class="QListWidget" name="lstRepresentation">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>151</width>
      <height>131</height>
     </rect>
    </property>
    <property name="currentRow">
     <number>1</number>
    </property>
    <item>
     <property name="text">
      <string>Tabular</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>RBF</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>IFDD</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="btnConfigRepresentation">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>参数设置</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_4">
   <property name="geometry">
    <rect>
     <x>560</x>
     <y>20</y>
     <width>171</width>
     <height>201</height>
    </rect>
   </property>
   <property name="title">
    <string>策略</string>
   </property>
   <widget class="QListWidget" name="lstPolicy">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>151</width>
      <height>131</height>
     </rect>
    </property>
    <property name="currentRow">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>eGreedy</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="btnConfigPolicy">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>参数设置</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_5">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>230</y>
     <width>711</width>
     <height>121</height>
    </rect>
   </property>
   <property name="title">
    <string>小车参数：</string>
   </property>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>30</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>小车位置(m)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>60</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>车速范围(m/s)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>90</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>施力范围(N)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>30</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>小车质量(kg)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>60</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>杆子质量(kg)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>90</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>杆子长度(m)：</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spPosition1">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>30</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-3.000000000000000</double>
    </property>
    <property name="maximum">
     <double>0.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>-2.400000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spPosition2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>30</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>3.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>2.400000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spVelocity1">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>60</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>0.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>-6.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spVelocity2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>60</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>6.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spAvailForce1">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>90</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-15.000000000000000</double>
    </property>
    <property name="maximum">
     <double>0.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
    <property name="value">
     <double>-10.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spAvailForce2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>90</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>15.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
    <property name="value">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spMassCart">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>30</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.500000000000000</double>
    </property>
    <property name="maximum">
     <double>1.500000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spMassPend">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.010000000000000</double>
    </property>
    <property name="maximum">
     <double>0.500000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spLength">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>90</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.500000000000000</double>
    </property>
    <property name="maximum">
     <double>1.500000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
